backend/kernelCI_app/management/commands/process_pending_aggregations.py

```diff
@@ -0,0 +1,394 @@
+import hashlib
+import time
+from typing import Sequence
+from django.core.management.base import BaseCommand
+from django.db import connection, transaction
+from kernelCI_app.constants.general import MAESTRO_DUMMY_BUILD_PREFIX
+from kernelCI_app.models import (
+    Builds,
+    Checkouts,
+    HardwareStatusEntityType,
+    PendingTest,
+    ProcessedHardwareStatus,
+    SimplifiedStatusChoices,
+    StatusChoices,
+)
+
+
+def get_hardware_key(origin: str, platform: str, checkout_id: str) -> bytes:
+    return hashlib.sha256(f"{origin}|{platform}|{checkout_id}".encode("utf-8")).digest()
+
+
+def calculate_status_count(status: str) -> tuple[int, int, int]:
+    if status in [StatusChoices.PASS, SimplifiedStatusChoices.PASS]:
```

Comment:
this is creating a list every time... either replace with 2 explicit comparison or use a tuple and declare it as a toplevel constant. Same for the case below below.

Also I'd stick with explicit `SimplifiedStatusChoices` values, if needed then you convert from `StatusChoices` to `SimplifiedStatusChoices` and then you call this function

Reply:
btw the name is also misleading, it's not calculating anything. And if you stick with a single choice you can do a mapping:

```py
simplified_status_choice_to_count = {
    SimplifiedStatusChoices.PASS: (1, 0, 0),
    SimplifiedStatusChoices.FAIL: (0, 1, 0),
    SimplifiedStatusChoices.INCOMPLETE: (0, 0, 1),
}
```